Author: Jan Dittberner <jandd@debian.org>
Subject: patch build.xml and native/Makefile to support Debian GNU/kFreeBSD

Index: libjna-java/native/Makefile
===================================================================
--- libjna-java.orig/native/Makefile	2014-06-10 23:48:11.333254681 +0200
+++ libjna-java/native/Makefile	2014-06-10 23:48:27.861046899 +0200
@@ -37,6 +37,7 @@
 OS=$(shell uname | sed -e 's/CYGWIN.*/win32/g' \
 	                -e 's/MINGW32.*/win32/g' \
                         -e 's/SunOS.*/solaris/g' \
+                        -e 's/GNU\/kFreeBSD/linux/g' \
                         -e 's/NetBSD/netbsd/g' \
                         -e 's/FreeBSD/freebsd/g' \
                         -e 's/OpenBSD/openbsd/g' \
Index: libjna-java/build.xml
===================================================================
--- libjna-java.orig/build.xml	2014-06-10 23:48:19.029157930 +0200
+++ libjna-java/build.xml	2014-06-10 23:48:27.861046899 +0200
@@ -169,6 +169,9 @@
     <condition property="os.prefix" value="freebsd-${os.arch}">
       <os name="FreeBSD"/>
     </condition>
+    <condition property="os.prefix" value="linux-${jre.arch}">
+      <os name="GNU/kFreeBSD"/>
+    </condition>
     <condition property="os.prefix" value="openbsd-${os.arch}">
       <os name="OpenBSD"/>
     </condition>
