Author: Jan Dittberner <jandd@debian.org>
Subject: patch build.xml and native/Makefile to support Debian GNU/kFreeBSD

Index: libjna-java/native/Makefile
===================================================================
--- libjna-java.orig/native/Makefile	2014-06-10 23:58:00.969841987 +0200
+++ libjna-java/native/Makefile	2014-06-11 00:09:03.021518829 +0200
@@ -39,6 +39,7 @@
 OS=$(shell uname | sed -e 's/CYGWIN.*/win32/g' \
 	                -e 's/MINGW32.*/win32/g' \
                         -e 's/SunOS.*/solaris/g' \
+                        -e 's/GNU\/kFreeBSD/linux/g' \
                         -e 's/NetBSD/netbsd/g' \
                         -e 's/FreeBSD/freebsd/g' \
                         -e 's/OpenBSD/openbsd/g' \
Index: libjna-java/build.xml
===================================================================
--- libjna-java.orig/build.xml	2014-06-11 00:08:53.785634940 +0200
+++ libjna-java/build.xml	2014-06-11 00:09:03.021518829 +0200
@@ -191,6 +191,9 @@
     <condition property="os.prefix" value="freebsd-${os.arch}">
       <os name="FreeBSD"/>
     </condition>
+    <condition property="os.prefix" value="linux-${jre.arch}">
+      <os name="GNU/kFreeBSD"/>
+    </condition>
     <condition property="os.prefix" value="openbsd-${os.arch}">
       <os name="OpenBSD"/>
     </condition>
