Author: Jan Dittberner <jandd@debian.org>
Subject: patch build.xml and native/Makefile to support Debian GNU/kFreeBSD

--- a/native/Makefile
+++ b/native/Makefile
@@ -37,6 +37,7 @@
 OS=$(shell uname | sed -e 's/CYGWIN.*/win32/g' \
 	                -e 's/MINGW32.*/win32/g' \
                         -e 's/SunOS.*/solaris/g' \
+                        -e 's/GNU\/kFreeBSD/linux/g' \
                         -e 's/NetBSD/netbsd/g' \
                         -e 's/FreeBSD/freebsd/g' \
                         -e 's/OpenBSD/openbsd/g' \
--- a/build.xml
+++ b/build.xml
@@ -169,6 +169,9 @@
     <condition property="os.prefix" value="freebsd-${os.arch}">
       <os name="FreeBSD"/>
     </condition>
+    <condition property="os.prefix" value="linux-${jre.arch}">
+      <os name="GNU/kFreeBSD"/>
+    </condition>
     <condition property="os.prefix" value="openbsd-${os.arch}">
       <os name="OpenBSD"/>
     </condition>
