Support all Debian multi-arch directories in library lookups.

Index: libjna-java/src/com/sun/jna/Platform.java
===================================================================
--- libjna-java.orig/src/com/sun/jna/Platform.java	2014-06-10 23:58:00.977841866 +0200
+++ libjna-java/src/com/sun/jna/Platform.java	2014-06-11 00:10:54.504116837 +0200
@@ -76,6 +76,12 @@
         else if (osName.equalsIgnoreCase("gnu/kfreebsd")) {
             osType = KFREEBSD;
         }
+        else if (osName.equalsIgnoreCase("gnu")) {
+            osType = GNU;
+        }
+        else if (osName.equalsIgnoreCase("gnu/kfreebsd")) {
+            osType = KFREEBSD;
+        }
         else {
             osType = UNSPECIFIED;
         }
@@ -139,6 +145,11 @@
     public static final boolean iskFreeBSD() {
         return osType == KFREEBSD;
     }
+    public static final boolean isGNU() {
+        return osType == GNU;
+    }    public static final boolean iskFreeBSD() {
+        return osType == KFREEBSD;
+    }
     public static final boolean isX11() {
         // TODO: check filesystem for /usr/X11 or some other X11-specific test
         return !Platform.isWindows() && !Platform.isMac();
@@ -195,4 +206,19 @@
             System.getProperty("os.arch").toLowerCase().trim();
         return arch.startsWith("arm");
     }
+
+    public static final String getBaseArch() {
+        String arch =
+            System.getProperty("os.arch").toLowerCase().trim();
+        if("amd64".equals(arch))
+            arch = "x86_64";
+        if("i686-at386".equals(arch))
+            arch = "i386";
+        if("ppc".equals(arch))
+            arch = "powerpc";
+        if("ppc64".equals(arch))
+            arch = "powerpc64";
+        return arch;
+    }
+
 }
